### PR TITLE
Reorganizes martial arts and stores them in the MIND.

### LINKED ISSUE #15918: martial art data not stored in mind
### What is the problem?

Learned martial arts (sleeping carp) stored as a variable in the body
### Why is it a problem, or what should have happened?

If you die and are cloned, you do not retain your martial art
### How can the problem be reproduced?

Kill and clone someone who has used a martial art scroll
### What revision did the problem happen on? (Helpful, but not required)

9d66dad13f54c5198d43d712c604f16bf80e93ac

### PR DESCRIPTION
**Pull request by ghost**
:cl: That Really Good Soda Flavor
fix: Martial arts are no longer lost when mind-swapping, cloning, moving your brain, et cetera.
/:cl:

They are stored totally in the mind now

Also, I moved martial arts to their own files like wrestling and krav maga have for some reason.

Fixes #15918

### CODE REVIEW

**Comment by Armhulen** [comment]
cool

**Comment by calzilla1** [comment]
Thank you jesus

**Comment by Jazaen** [comment]
Could you do both?
As in, both keeping them in body and giving them to mind?

**Comment by Armhulen** [comment]
Wait, does this mean changelings will learn martial arts when they succ someone?

**Comment by optimumtact** [comment]
Who downthumbs this lol

**Comment by Jazaen** [comment]
You can hover over it to see memefruit.
I know you know this.

**Comment by Cyberboss** [comment]
Ugh, ignore my review, martial is full of copypasta shitcode that needs a different PR to refactor. My main complaint is this: You can't go halfway on this, either fully store them in the mind or don't move it at all.

**Comment by Really-Good-Soda-Flavor** [comment]
I went the whole way now

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1140,7 +1140,7 @@
 /datum/species/proc/spec_hitby(atom/movable/AM, mob/living/carbon/human/H)
 	return
 
-/datum/species/proc/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style = M.martial_art)
+/datum/species/proc/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style = M.mind.martial_art)
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species.dm]: Mind is not guaranteed to exist here. This should default to some global/mob static and be passed explicitly in the call.

**Comment by Tacolizard** [comment]
KNAWLEDGE

**Comment by Really-Good-Soda-Flavor** [comment]
[This is not a feature]

**Comment by AnturK** [comment]
Autotagger gone berserk.

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 2]
```diff
@@ -406,7 +406,9 @@
 		else
 			reactive_teleport(H)
 
-/datum/species/golem/bluespace/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style = M.martial_art)
+/datum/species/golem/bluespace/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style)
+	if(H.mind)
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: H > M (Also these could use renaming but that's for another PR)

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 3]
```diff
@@ -490,7 +492,9 @@
 	var/golem_name = "[uppertext(clown_name)]"
 	return golem_name
 
-/datum/species/golem/bananium/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style = M.martial_art)
+/datum/species/golem/bananium/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style)
+	if(H.mind)
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: Same here

File: code/modules/mob/living/carbon/human/species.dm [hunk 4]
```diff
@@ -1140,14 +1140,16 @@
...
-/datum/species/proc/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style = M.martial_art)
+/datum/species/proc/spec_attack_hand(mob/living/carbon/human/M, mob/living/carbon/human/H, datum/martial_art/attacker_style)
 	if(!istype(M))
 		return
 	CHECK_DNA_AND_SPECIES(M)
 	CHECK_DNA_AND_SPECIES(H)
 
 	if(!istype(M)) //sanity check for drones.
 		return
+	if(H.mind)
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species.dm]: H > M. Could the arguments be removed if these are checked at this stage anyway ?

**Comment by Cyberboss** [comment]
Its based on his original changelog, so he must've had `add:` in there first

File: code/modules/mob/living/carbon/human/species_types/golems.dm (hunk 2 above)
Comment (Really-Good-Soda-Flavor) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: Please be more specific on what you mean?

Comment (AnturK) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: You check if defender minds exists then access attackers mind.

**Comment by Really-Good-Soda-Flavor** [comment]
Any other places that need fixing or new sanity checks?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**